luigi/contrib/snowflake.py

```diff
@@ -0,0 +1,523 @@
+import abc
+import logging
+
+import luigi
+
+from enum import Enum
+from luigi import six
+from luigi.contrib import postgres
+from luigi.contrib import rdbms
+
+logger = logging.getLogger('luigi-interface')
+
+# Snowflake's library does not have constant for their error code.
+# We define our own
+SNOWFLAKE_TABLE_EXISTS_ERROR = 2002
+SNOWFLAKE_GENERIC_PROGRAMMING_ERROR = 2003
+
+try:
+    import snowflake
+    import snowflake.connector
+except ImportError:
+    logger.warning("Loading snowflake module without snowflake-connector-python installed. "
+                   "Will crash at runtime if snowflake functionality is used.")
+
+
+class SnowflakeSupportedAuth(Enum):
+    default = 0
+    aws_key_auth = 0
+    # next value = 1
+
+    @classmethod
+    def get(cls, which):
+        if which == cls.aws_key_auth:
+            return AwsKeyAuth()
+        else:
+            raise ValueError("SnowflakeSupportedAuth value ' {0} ' isn't supported", which)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class SnowflakeAuth(object):
+    """Abstractable SnowflakeAuth base class that can be replace by tool
+    specific implementation."""
+    @property
+    def configuration_section(self):
+        """
+        Override to change the configuration section used
+        to obtain default credentials.
+        """
+        return 'snowflake'
+
+    @abc.abstractmethod
+    def credentials(self, task):
+        pass
+
+
+class _SettingsMixins(object):
+    """
+    Generic mixins that defines how to fetch the database and warehouse
+    configuration for a given snowflake integration.
+    """
+    @property
+    def configuration_section(self):
+        """
+        Override to change the configuration section used
+        to obtain default credentials.
+        """
+        return 'snowflake'
```

Comment:
what is the motivation behind allowing both the` _SettingsMixin` and `SnowflakeAuth` to separately set their `configuration_section`s? Why would you want them to differ?